server/controllers/routes/teacherProfile.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require('jsonwebtoken');
+const { getTeacherProfile } = require('../../database/quieres/teacherProfile');
+
+require('env2')('./config.env');
+
+const secret = process.env.SECRET_KEY;
+
+const teacherProfile = (req, res, next) => {
+  getTeacherProfile()
+    .then(({ rows }) => {
+      if (req.cookies.access) {
```

Comment:
why are you getting around the `auth` middleware?
Here you first get the data from the DB and then verify the cookie - that's wrong, should be the other way around. And with this approach you would have to repeat this code in every API that doesn't use the `auth` middleware, so in the very least it should have been extracted into a function, but using the middleware is the correct approach, this doesn't look right.

Reply:
I got you , and I have edit the code correctly